### PR TITLE
Update dependencies and APIs for AGP 7

### PR DESCRIPTION
**Pull request by ZacSweers**

### CODE REVIEW

File: sample-libraries/a/build.gradle [hunk 1]
```diff
@@ -18,14 +18,14 @@ apply plugin: 'com.android.library'
...
+  compileSdk 31
   defaultConfig {
     minSdkVersion 21
   }
 
   compileOptions {
-    sourceCompatibility = JavaVersion.VERSION_1_8
-    targetCompatibility = JavaVersion.VERSION_1_8
+    sourceCompatibility = JavaVersion.VERSION_11
+    targetCompatibility = JavaVersion.VERSION_11
```
Comment (jschear) [inline sample-libraries/a/build.gradle:28]: Not that it matters much for these samples, but shouldn't source and target compatibility for applications stay at Java 8?

**Review by jschear — APPROVED** [review]

File: sample-libraries/a/build.gradle (hunk 1 above)
Comment (ZacSweers) [reply sample-libraries/a/build.gradle:28]: Not necessary for the sample!

Comment (jschear) [reply sample-libraries/a/build.gradle:28]: Oh man, I hadn't realized that Java 11 was already supported for source/target compatibility (through D8 I guess?).

Comment (ZacSweers) [reply sample-libraries/a/build.gradle:28]: Yup!

**Pull request merged**